NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,228 @@
+        let button = buttonsArray.filter{$0.tag == selectedSegmentIndex}.first
+        
+        selectedSegmentIndex = sender.tag
+        
+        if let previousButton = button {
+            if previousButton.tag != sender.tag {
+                sendActions(for: .valueChanged)
+                previousButton.isSelected = false
+                UIButton.animate(withDuration: 0.3, animations: {
+                    previousButton.transform = CGAffineTransform.identity
+                })
+            }
+        }
+        
+        sender.isSelected = true
+    }
+    
+    // Private
+    // Set up
+    private func setUp() {
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
+    }
+    
+    // Create buttons
+    private func createButtons() {
+        for button in buttonsArray {
+            button.removeFromSuperview()
+        }
+        buttonsArray.removeAll()
+        
+        for (index, title) in buttonTitles.enumerated() {
+            let button = UIButton(type: .custom)
+        
+            button.setTitleColor(UIColor.lightGray, for: .normal)
+            button.setTitleColor(UIColor.black, for: .selected)
+            button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+            button.setTitle(title, for: .normal)
+            
+            button.tag = index
+            
+            buttonsArray.append(button)
+            
+            scrollView.addSubview(button)
+        }
+    }
+    
+    private func updateUI() {
+        var buttonX: CGFloat = buttonMargin
+        var contentSizeWidth: CGFloat = 0
+        
+        // Set button frame
+        for button in buttonsArray {
+            button.transform = CGAffineTransform.identity
+            button.titleLabel?.font = font
+            
+            let buttonSize = calculateButtonSize(title: (button.titleLabel?.text)!, fontAttributes: [NSAttributedStringKey.font: font!])
+            
+            button.frame = CGRect(x: buttonX,
+                                  y: 0,
+                                  width: buttonSize.width,
+                                  height: scrollView.frame.size.height)
+            
+            buttonX += button.frame.size.width + buttonSpacing
+            
+            contentSizeWidth += button.frame.size.width
+        }
+        
+        scrollView.contentSize.width = buttonX
+    }
+    
+    func calculateButtonSize(title: String, fontAttributes: [NSAttributedStringKey : Any]) -> CGSize {
+        return title.size(withAttributes: fontAttributes)
+    }
+    
+    // Draw separator line
+    private func drawSeparatorLine() {
+        let separatorHeight: CGFloat = 1
+
+        let separator = UIView(frame: CGRect(x: 0,
+                                             y: frame.size.height - separatorHeight,
+                                             width: frame.size.width,
+                                             height: separatorHeight))
```

Comment:
It's good that you now moved drawing of separator line out of `layoutSubviews()`, but you should still set its frame in `layoutSubviews()` method.

The issue with that is not occurring right now, since you're using Storyboards. But if you created your control from code, the separator would've been invisible, because of the `frame` being `CGRect.zero` at this point.